atom/browser/api/atom_api_top_level_window.cc

```diff
@@ -398,8 +398,11 @@ gfx::Rect TopLevelWindow::GetContentBounds() {
 
 void TopLevelWindow::SetSize(int width, int height, mate::Arguments* args) {
   bool animate = false;
-  args->GetNext(&animate);
-  window_->SetSize(gfx::Size(width, height), animate);
+  gfx::Size size = window_->GetMinimumSize();
+  if ((size.height() <= height) | (size.width() <= width)) {
```

Comment:
it'd be better to snap to the smallest dimension rather than just fail to resize completely. e.g.

```cpp
size.SetToMax(gfx::Size(width, height));
```